scripts/functions/utils.py

```diff
@@ -0,0 +1,47 @@
+#----------------------------------------------------------------------------------------------------------------
+# Helper functions
+#----------------------------------------------------------------------------------------------------------------
+import pandas as pd
+import numpy as np
+import os
+
+def get_scale_factor (ref_file, gene_id, expression_profile):
+    
+
+    """
+    Determine how much to scale offset based on distance along the target gene expression gradient
+    
+    Output:
+     scale factor = intermediate gene expression/ (average high target gene expression - avgerage low target gene expression) 
+    """
+
+    # Read in reference gene expression file
+    ref_data = pd.read_table(ref_file, header=0, sep='\t', index_col=0)
```

Comment:
i noticed this is being called in all the loops. Instead of reading in the data each time, why not include `ref_data` as a function argument and read in before looping

Reply:
This is a good point, I have changed this to read in the objected that is already read in instead of read it in each time through the for loop